Services/UI/OJS.Services.Ui.Business/Implementations/SubmissionsBusinessService.cs

```diff
@@ -99,6 +104,26 @@ public async Task<SubmissionDetailsServiceModel> GetDetailsById(int submissionId
             throw new BusinessServiceException(validationResult.Message);
         }
 
+        var contest = await this.contestsDataService.GetByProblemId<ContestServiceModel>(submissionDetailsServiceModel!.Problem.Id).Map<Contest>();
+        var userIsAdminOrLecturerInContest = currentUser.IsAdmin || IsUserLecturerInContest(contest, currentUser.Id!);
+        var showTestInputForAllTests = submissionDetailsServiceModel.Problem.ShowDetailedFeedback;
+        if (!userIsAdminOrLecturerInContest && !showTestInputForAllTests)
+        {
+            submissionDetailsServiceModel.TestRuns = submissionDetailsServiceModel.TestRuns.Select(x =>
```

Comment:
In order the code to be more readable, you can change this to `.Select(tr =>
            {
                if (!tr.IsTrialTest)
                {
                    tr.ShowInput = false;
                    tr.Input = string.Empty;
                    tr.ExecutionComment = string.Empty;
                    tr.ExpectedOutputFragment = string.Empty;
                    tr.UserOutputFragment = string.Empty;
                }
                return tr;
            });
        }`

Reply:
Done.